google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionStatementExecutorTest.java

```diff
@@ -237,4 +251,290 @@ public void testStatementSetPgTransactionModeNoOp() {
     verify(connection, never()).setTransactionMode(TransactionMode.READ_ONLY_TRANSACTION);
     verify(connection, never()).setTransactionMode(TransactionMode.READ_WRITE_TRANSACTION);
   }
+
+  private ResultSet getMockResultSetForAnalyseQuery(){
```

Comment:
Use `Mockito` to create a mock for this instead. See here for an example how to do that: https://github.com/googleapis/java-spanner/blob/12a0acb02785987d8442c03a45b8b04a068375d7/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionImplTest.java#L196